### ISSUE TITLE
Sufficient Number of Audio Files for LoRA Trainig?

### ISSUE THREAD

**Issue by mdicillo** [woct0rdho/ACE-Step#17]
Hello!

This isn't really an issue, but a question for others out there training with this method.

How many audio files are you using as your dataset and what are the lengths of those files?

I have very small dataset since I am training on a fairly niche genre that varies in length from 45 - 90 seconds. The amount of files varies from 20 to 40 tracks.

I have already tagged the audio files myself since these are part of an existing catalog and am using those tags for the prompt file. These are also all instrumental pieces, so each lyric file is just simply "[instrumental]."

I am definitely hearing the effects of the training, but am wondering if my results are limited by the amount of training material.

Does anyone else have any insight and willing to share their dataset sizes and possible epoch/steps numbers?

Here are mine:
**Number of tracks: 20 - 40
Length: 45 - 90
Vocals: No - All Instrumental
Epochs: 0
Max Steps: 2000**

I notice with smaller datasets, it's hard to increase the number of steps as it eventually leads to overfitting. Is there another method that could avoid this or at least minimize the overfitting?

**Comment by woct0rdho** [comment]
If you can't find more audio, I guess you can do some data augmentation. An easy way of augmentation is to slightly change the volume.

**Comment by woct0rdho** [comment]
You can also try this but the generated audio may become a chunk rather than a full song.

**Comment by mdicillo** [comment]
This would actually be fine in my case since I'm not looking to generate full songs. I am aiming for instrumental tracks in the 45 - 60 second range, so this may work to my benefit!

Do you have any input regarding the --max_steps and --epoch values, or are these usually determined on a case-by-case basis after the analysis?

**Comment by woct0rdho** [comment]
To start with, you can set `max_steps` to very large and `epochs` to -1 (no epoch limit), then monitor the loss graph. When the loss stops decreasing, that means the number of steps is enough.
